client/src/components/Donation/PaypalButton.js

```diff
@@ -41,8 +41,8 @@ export class PaypalButton extends Component {
     const { amount, duration } = this.state;
     const { skipAddDonation = false } = this.props;
 
-    // Skip the api if user is not signed in or if its a one-time donation
-    if (!skipAddDonation || isSubscription) {
+    // Skip the api if user is not signed in and if it is a one-time donation
+    if (!skipAddDonation && isSubscription) {
```

Comment:
```suggestion
    if (isSignedIn && isSubscription) {
    // call the api
```
Bit nitpicky, but the comment and the logic were in conflict.  Also, I think this is a fraction easier to understand.

Reply:
I would also prefer this, as it is not immediately obvious when and why `skipAddDonation` is used.

Reply:
Done